src/analyses/goto_rw.h

```diff
@@ -60,12 +68,27 @@ inline range_spect to_range_spect(const mp_integer &size)
 }
 
 // each element x represents a range of bits [x.first, x.second.first)
-class range_domaint:
-  public range_domain_baset,
-  public std::list<std::pair<range_spect, range_spect> >
+class range_domaint:public range_domain_baset
 {
+  using sub_typet=std::list<std::pair<range_spect, range_spect>>;
+  sub_typet data_;
+
 public:
-  virtual void output(const namespacet &ns, std::ostream &out) const;
+  void output(const namespacet &ns, std::ostream &out) const override;
+
+  using iterator=sub_typet::iterator;
+  using const_iterator=sub_typet::const_iterator;
```

Comment:
None, but `using` is more versatile than `typedef` so should be preferred in general. I know we don't do this yet, but at some point I'll submit a PR to convert `typedef` to `using` so that everything's consistent.